python/ray/autoscaler/v2/tests/test_node_provider.py

```diff
@@ -610,6 +610,112 @@ def test_inconsistent_pods_raycr_scale_down(self):
+
+        # Identify all pods in the target group
+        small_group = "small-group"
+        pod_names = []
+        for pod in self.mock_client._pod_list["items"]:
+            if pod["metadata"]["labels"]["ray.io/group"] == small_group:
+                pod_names.append(pod["metadata"]["name"])
+
+        # Terminate all pods in the group
+        self.provider._sync_with_api_server()
+        cur_instance_ids = set(self.provider.instances.keys())
+        pods_to_terminate = [name for name in pod_names if name in cur_instance_ids]
+
+        assert (
+            len(pods_to_terminate) > 1
+        ), "Expected multiple pods to terminate in the target group."
+
+        self.provider.terminate(ids=pods_to_terminate, request_id="term-2")
+
+        # Check the patches applied to the RayCluster resource
+        patches = self.mock_client.get_patches(
+            f"rayclusters/{self.provider._cluster_name}"
+        )
+
+        assert len(patches) == 2
+        assert patches == [
+            {
+                "op": "replace",
+                "path": "/spec/workerGroupSpecs/0/replicas",
+                "value": 0,
+            },
+            {
+                "op": "replace",
+                "path": "/spec/workerGroupSpecs/0/scaleStrategy",
+                "value": {
+                    "workersToDelete": pods_to_terminate,
+                },
+            },
+        ]
+
+    def test_worker_to_delete_info(self):
+        """
+        Validate _get_workers_delete_info correctly returns the worker groups with pending
+        deletes, worker groups with finished deletes, and the set of workers to delete.
+        """
+        # Setup provider with multiple worker pods in podlist. We set replicas to 0
+        # to simulate the case where the autoscaler patches the RayCluster `replicas: 0`
+        # but alive Pods still exist in workersToDelete.
+        raycluster_cr = get_basic_ray_cr()
+        raycluster_cr["spec"]["workerGroupSpecs"][0]["replicas"] = 0
+        self.mock_client = MockKubernetesHttpApiClient(
+            _get_test_yaml("podlist2.yaml"), raycluster_cr
+        )
+        self.provider = KubeRayProvider(
+            cluster_name="test",
+            provider_config={
+                "namespace": "default",
+                "head_node_type": "headgroup",
+            },
+            k8s_api_client=self.mock_client,
+        )
+
+        # Add some workers to workersToDelete.
+        small_group = "small-group"
+        pod_names = []
+        for pod in self.mock_client._pod_list["items"]:
+            if pod["metadata"]["labels"]["ray.io/group"] == small_group:
+                pod_names.append(pod["metadata"]["name"])
+        raycluster_cr["spec"]["workerGroupSpecs"][0]["scaleStrategy"] = {
+            "workersToDelete": pod_names,
+        }
+
+        (
+            pending_deletes,
+            finished_deletes,
+            workers_to_delete,
+        ) = self.provider._get_workers_delete_info(
```

Comment:
We just had a conversation about not testing private method (i.e. implementation detail): https://github.com/ray-project/ray/pull/52812#discussion_r2085325706. Can we test the public interface instead?

Reply:
Let's chat in-person.